mlrun/utils/helpers.py

```diff
@@ -1718,6 +1718,91 @@ def validate_single_def_handler(function_kind: str, code: str):
             )
 
 
+def validate_node_selectors(
+    node_selectors: dict[str, str], raise_on_error: bool = True
+) -> bool:
+    """
+    Ensures that user-defined node selectors adhere to Kubernetes label standards:
+    - Validates that each key conforms to Kubernetes naming conventions, with specific rules for name and prefix.
+    - Ensures values comply with Kubernetes label value rules.
+    - If raise_on_error is True, raises errors for invalid selectors.
+    - If raise_on_error is False, logs warnings for invalid selectors.
+    """
+    # Define regex patterns
+    label_pattern = re.compile(mlrun.utils.regex.k8s_label_pattern)
+    prefix_pattern = re.compile(mlrun.utils.regex.k8s_label_prefix_pattern)
+
+    # Helper function for handling errors or warnings
+    def handle_invalid(message):
+        if raise_on_error:
+            raise mlrun.errors.MLRunInvalidArgumentError(message)
+        else:
+            warnings.warn(
+                f"{message} \nFor your current SDK version, "
+                f"the node selector you've set does not comply with the validation rules. "
+                f"Check our documentation for the latest validation updates. "
+                f"If there's a newer validation standard, consider upgrading. "
+                f"Be aware that your function executions may fail if the node selector remains invalid."
+            )
```

Comment:
The validation failed not because the SDK version is old, it's because the node selector doesn't adhere to **k8s'** naming rules.
I wouldn't go into suggesting "try that, or maybe this, but be careful". It failed because it's not valid, and that's it.